shell/models/fleet.cattle.io.cluster.js

```diff
@@ -124,19 +125,55 @@ export default class FleetCluster extends SteveModel {
     return mgmt;
   }
 
-  get norman() {
-    const norman = this.$rootGetters['rancher/byId'](NORMAN.CLUSTER, this.metadata.labels?.[FLEET_LABELS.CLUSTER_NAME]);
+  get basicNorman() {
+    const norman = this.$rootGetters['rancher/byId'](NORMAN.CLUSTER, this.metadata?.labels?.[FLEET_LABELS.CLUSTER_NAME]);
 
     return norman;
   }
 
+  get norman() {
+    if (this.basicNorman) {
+      return this.basicNorman;
+    }
+
+    // If navigate to YAML view directly, norman is not loaded yet
+    return (async() => {
+      return await this.$dispatch('rancher/find', { type: NORMAN.CLUSTER, id: this.metadata.labels[FLEET_LABELS.CLUSTER_NAME] }, { root: true });
+    })();
+  }
+
+  async normanClone() {
+    const norman = await this.norman;
+
+    const nc = await this.$dispatch('rancher/clone', { resource: norman }, { root: true });
+
+    if (!nc.metadata) {
+      nc.metadata = {};
+    }
+
+    return nc;
+  }
+
   get groupByLabel() {
     const name = this.metadata.namespace;
 
-    if ( name ) {
+    if (name) {
       return this.$rootGetters['i18n/t']('resourceTable.groupLabel.workspace', { name: escapeHtml(name) });
     } else {
       return this.$rootGetters['i18n/t']('resourceTable.groupLabel.notInAWorkspace');
     }
   }
+
+  async saveYaml(yaml) {
+    await this._saveYaml(yaml);
+
+    const parsed = jsyaml.load(yaml);
+
+    const norman = await this.normanClone();
+
+    norman.labels = { ...parsed.metadata.labels };
```

Comment:
classes that inherit from the `NormanModel` have setLabels and setAnnotations methods which should be used here